protocol/metrics/consumer_relayserver_client.go

```diff
@@ -51,7 +52,7 @@ func NewConsumerRelayserverClient(endPointAddress string) *ConsumerRelayserverCl
 	return cuc
 }
 
-func (cuc *ConsumerRelayserverClient) processQueue() {
+func (cuc *ConsumerRelayServerClient) processQueue() {
```

Comment:
refactor this method, its kinda. dead locks can kill our service so that's why I'm very specific on how it should look. we had too many in the past and its hell to bebug 

create an inner method. 
```go 
func (cuc *ConsumerRelayServerClient) shouldSendInfo() bool {
    cuc.lock.RLock()
    defer cuc.lock.RUnlock()
    return !cuc.isSendQueueRunning && len(cuc.addQueue) > 0
}
```

now on main flow:

if cuc.shouldSendInfo() {
    ....
    ....
} 

every time you need to read do an rlock with a defer 
every time you need to write lock. 

if you need a flow with multiple reads and write just lock the entire flow. 

best practice is a method starting with lock + defer unlock and do what ever you need inside.